test/src/kbasesearchengine/test/common/FileUtilTest.java

```diff
@@ -0,0 +1,63 @@
+package kbasesearchengine.test.common;
+
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+
+import kbasesearchengine.common.FileUtil;
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.Test;
+import org.junit.BeforeClass;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+
+/**
+ * Created by apasha on 12/13/17.
+ */
+public class FileUtilTest {
+    private static Path tempDirPath;
+    private static final String subDirName = "testSubDir";
+
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        tempDirPath = Paths.get(TestCommon.getTempDir()).resolve("FileUtilTest");
+        tempDirPath.toFile().mkdirs();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        FileUtils.deleteDirectory(tempDirPath.toFile());
+    }
+
+    @Test
+    public void getOrCreateSubDirTest() {
+        File subDir = FileUtil.getOrCreateSubDir(tempDirPath.toFile(), subDirName);
```

Comment:
Should test that the expected directory was returned

Reply:
I meant that with this test, the fn could be returning any old directory and ignoring subDirName and tempDirPath.toFile(), and the test woudn't catch it. The test should check the contents of subDir to ensure they're as expected.